src/ldp/nn/agent/simple_local_agent.py

```diff
@@ -42,6 +46,10 @@ class AgentLMConfig(_LMConfig):
         ),
         validate_default=True,
     )
+    max_messages_token_count: int | None = Field(
+        default=None,
+        description="If set, raise an error if the total tokens in the trajectory exceed this value.",
```

Comment:
```suggestion
        description="If set, raise an error if the total tokens in the message history and tool description exceed this value.",
```

Since it includes tools, maybe we should call it `max_token_count` - I just don't want to call it a trajectory, since in principle it is just (state, observation) [e.g. for SeqQA we don't pass the full trajectory]